orc8r/cloud/go/services/state/types/types.go

```diff
@@ -30,26 +41,56 @@ const (
 	SerdeDomain = "state"
 )
 
-// StatesByID maps state IDs to state.
-// A state and its ID collectively contains all information for a piece of state.
-type StatesByID map[ID]State
-
 // ID identifies a piece of state.
-// A piece of state is uniquely identified by the triplet {network ID, device ID, type}.
+// A piece of state is uniquely identified by the triplet
+// {network ID, device ID, type}.
 type ID struct {
-	// Type determines how the value is deserialized and validated on the cloud service side.
+	// Type determines how the value is deserialized and validated.
 	Type string
-	// DeviceID is the ID of the entity with which the state is associated (IMSI, serial number, etc).
+	// DeviceID is the ID of the entity with which the state is associated
+	// (IMSI, serial number, etc).
 	DeviceID string
 }
 
-// State includes reported operational state and additional info about the reporter.
+// IDs are a list of state IDs.
+type IDs []ID
+
+// MakeIDs converts a list of device IDs to a list of state IDs.
+func MakeIDs(typ string, keys ...string) IDs {
+	var ids IDs
+	for _, key := range keys {
+		ids = append(ids, ID{Type: typ, DeviceID: key})
+	}
+	return ids
+}
+
+// IDsByNetwork are a set of state IDs, keyed by network ID.
+type IDsByNetwork map[string]IDs
+
+// SerializedState directly wraps serialized reported state with metadata.
+type SerializedState struct {
```

Comment:
Why does this have `ReportedID` and `TimeMs` but not `State`? I also think this is probably an unnecessary addition to the API like the entity version

Reply:
The type already existed in the pkg, it just wasn't exposed via the client api. From offline, we'll keep the SerializedState obj but refrain from exposing it to the client api

Reply:
After playing around with it, the SerializedState is only leaked to callers from within the state service (state indexer), so planning to leave it as-is